src/components/app/Home.js

```diff
@@ -49,13 +52,29 @@ class InstallButton extends React.PureComponent<InstallButtonProps> {
   }
 }
 
-type UploadButtonProps = {
+type UploadButtonsProps = {
   retrieveProfileFromFile: typeof retrieveProfileFromFile,
+  triggerLoadingFromUrl: typeof triggerLoadingFromUrl,
 };
 
-class UploadButton extends React.PureComponent<UploadButtonProps> {
+type UploadButtonsState = {
+  isLoadFromUrlPressed: boolean,
+};
+
+type UploadFromUrlProps = {
+  triggerLoadingFromUrl: typeof triggerLoadingFromUrl,
+};
+
+class UploadButtons extends React.PureComponent<
```

Comment:
nit: please call the class `ActionButtons`, because they're not all for "upload".